driver/restdriver.go

```diff
@@ -37,19 +52,422 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
+
+	driver.logger.Debugf("In HandleReadCommands function")
+	var reading interface{}
+	var val interface{}
+	var result = &dsModels.CommandValue{}
+	var uri string
+
+	// Declare response structure,
+	// This will be filled with response received from HTTP GET request
+	responses = make([]*dsModels.CommandValue, len(reqs))
+
+	// To send request to any end device, first we need to know end device details.
+	// Such as end device IP address, port number on which REST server is running etc.
+	// First get all these details from the device file
+	protocolParams, paramsExists := protocols[ED_PARAMS]
+	if !paramsExists {
+		return nil, fmt.Errorf("No End device parameters defined in the protocol list")
+	}
+
+	// Get end device IP address
+	address, ok := protocolParams[ED_IP]
+	if !ok {
+		return nil, fmt.Errorf("ED_IP not found")
+	}
+	driver.logger.Debugf("ED_IP:%v", address)
+
+	// Get end device port number
+	port, ok := protocolParams[ED_PORT]
+	if !ok {
+		return nil, fmt.Errorf("ED_PORT not found")
+	}
+	driver.logger.Debugf("ED_PORT:%v", port)
+
+	// Get end device URI prefix, This parameter will be empty for the end devices
+	// which do not have any prefix
+	uriPrefix, ok := protocolParams[ED_URI_PREFIX]
+	if !ok {
+		return nil, fmt.Errorf("ED_URI_PREFIX not found")
+	}
+	driver.logger.Debugf("ED_URI_PREFIX:%v", uriPrefix)
+
+	// Now, we have got required end device information, its time to create GET request
+	for i, req := range reqs {
+		// First get device resource instance, needed during validation of the
+		// response data received from the end device later.
+		// RunningService returns the Service instance which is running.
+		// service.DeviceResource retrieves the specific DeviceResource instance
+		// from cache according to the Device name and Device Resource name
+		resourceName := req.DeviceResourceName
+
+		service := sdk.RunningService()
+
+		deviceResource, ok := service.DeviceResource(deviceName, resourceName)
+		if !ok {
+			driver.logger.Errorf("Incoming reading ignored. Resource '%s' not found", resourceName)
+			return nil, fmt.Errorf("Resource not found")
+		}
+
+		// Now get the query parameters received in the request.
+		// These needs to be sent to the end device
+		// Query parameters needs to be converted to string to append to the uri
+		driver.logger.Debugf("URLRawQuery :%v", req.Attributes[URLRawQuery])
+		reqParam := fmt.Sprint(req.Attributes[URLRawQuery])
+
+		// Form URI from the end device parameters and request parameters and
+		// query parameters. Omit uri prefix if it is empty
+		if uriPrefix != "" {
+			uri = "http://" + address + ":" + port + "/" + uriPrefix + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		} else {
+			uri = "http://" + address + ":" + port + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		}
+		driver.logger.Debugf("uri = %v", uri)
```

Comment:
This is the one debug statement need. the remaining can be removed.
```suggestion
		driver.logger.Debugf("Sending REST Get command to uri = %v", uri)
```